gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -0,0 +1,231 @@
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageImpl;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageItemInfo;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage.StorageOperation;
+import com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.TestBucketHelper;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.EnumSet;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GoogleCloudStorageGrpcIntegrationTest {
+
+  // This prefix will be replaced by the prefix used in other gcs io integrate tests once it's
+  // whitelisted by GCS to access gRPC API.
+  private static final String BUCKET_NAME_PREFIX = "gcs-grpc-team-dataproc-it";
+
+  private static final Supplier<TestBucketHelper> BUCKET_HELPER =
+      Suppliers.memoize(() -> new TestBucketHelper(BUCKET_NAME_PREFIX));
+
+  private static final LoadingCache<GoogleCloudStorage, String> SHARED_BUCKETS =
+      CacheBuilder.newBuilder()
+          .build(
+              new CacheLoader<GoogleCloudStorage, String>() {
+                @Override
+                public String load(GoogleCloudStorage gcs) throws Exception {
+                  return BUCKET_HELPER.get().getUniqueBucketName("shared") + "_" + gcs.hashCode();
+                }
+              });
+
+  private final GoogleCloudStorage rawStorage;
+
+  public GoogleCloudStorageGrpcIntegrationTest() throws IOException {
+    GoogleCloudStorage gcs =
+        new ThrottledGoogleCloudStorage(
+            // Allow 2 create or delete bucket operation every second. This will hit rate limits,
+            // but GCS now has back-offs implemented for bucket operations.
+            RateLimiter.create(2),
+            new GoogleCloudStorageImpl(
+                GoogleCloudStorageTestHelper.getStandardOptionBuilder()
+                    .setGrpcEnabled(true)
+                    .build(),
+                GoogleCloudStorageTestHelper.getCredential()),
+            EnumSet.of(StorageOperation.DELETE_BUCKETS, StorageOperation.CREATE_BUCKET));
+    this.rawStorage = gcs;
```

Comment:
Tests initialize their setup usually in `@Before` method, not constructor.

Reply:
Moved this into a static filed to make sure it's only created once. (constructor itself will be called as many times as number of tests.